### PR TITLE
Jakeob Ewell completed Express notes api

### PR DESCRIPTION
**Pull request by jakeobewell**
express-notes-api

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  const datajson = require('./data.json');
+  const notes = datajson.notes
+  for (const id in notes) {
+    notesArr.push(notes[id]);
+  }
+  res.status(200).json(notesArr);
+})
+
+app.get('/api/notes/:id', (req, res) => {
+  const datajson = require('./data.json');
```
Comment (Zhiwutian) [inline express-notes-api/index.js:21]: make sure all of your require statements occur at the top of the document

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  const noteId = req.params.id
+  const notes = datajson.notes
+  const note = notes[noteId];
+  if (`${parseInt(noteId)}` === 'NaN' || parseInt(noteId) < 1) {
```
Comment (Zhiwutian) [inline express-notes-api/index.js:25]: nothing is every === to NaN, you should be checking if the number is not an integer here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
check out the isInteger method

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    })
+  }
+})
+
+app.delete('/api/notes/:id', (req, res) => {
+  const fs = require('fs');
+  const datajson = require('./data.json');
+  const noteId = req.params.id;
+  const note = datajson.notes[noteId];
+  delete datajson.notes[noteId];
```
Comment (Zhiwutian) [inline express-notes-api/index.js:76]: don't delete before you confirm that the property exsits

File: express-notes-api/index.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+  if (`${parseInt(noteId)}` === 'NaN' || parseInt(noteId) < 1) {
```
Comment (Zhiwutian) [inline express-notes-api/index.js:78]: same issue as above, nothing is === to NaN

File: express-notes-api/index.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+app.put('/api/notes/:id', (req, res) => {
+  const fs = require('fs');
+  const datajson = require('./data.json');
+  const noteId = req.params.id;
+  const originalNote = datajson.notes[noteId];
+  const newNote = req.body;
+  newNote.id = parseInt(noteId);
+
+  if (`${parseInt(noteId)}` === 'NaN' || parseInt(noteId) < 1) {
```
Comment (Zhiwutian) [inline express-notes-api/index.js:113]: same as other integer checks

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
looking great, just a few updates!

File: express-notes-api/index.js (hunk 2 above)
Comment (jakeobewell) [reply express-notes-api/index.js:25]: Yah I figured out in a previous challenge that nothing is equal to NaN that's why if it is NaN it comes out as a string "NaN" and I am checking if it is equal to that string. It works, but if you think isInteger is better I can change it.

**Review by Zhiwutian — APPROVED** [review]
👍

**Pull request merged**